tests/regression/tools/notification/base_client.c

```diff
@@ -0,0 +1,404 @@
+#include <lttng/endpoint.h>
+#include <lttng/notification/channel.h>
+#include <lttng/notification/notification.h>
+#include <lttng/trigger/trigger.h>
+
+static unsigned int nr_notifications = 0;
+static unsigned int nr_expected_notifications = 0;
+static const char *session_name = NULL;
+static const char *channel_name = NULL;
+static double threshold_ratio = 0.0;
+static uint64_t threshold_bytes = 0;
+static bool is_threshold_ratio = false;
+static enum lttng_condition_type buffer_usage_type = LTTNG_CONDITION_TYPE_UNKNOWN;
+static enum lttng_domain_type domain_type = LTTNG_DOMAIN_NONE;
+
+int handle_condition(
+		const struct lttng_condition *condition,
+		const struct lttng_evaluation *condition_evaluation);
+
+int parse_arguments(char **argv) {
+	const char *domain_type_string = NULL;
+	const char *buffer_usage_type_string = NULL;
+	const char *buffer_usage_threshold_type = NULL;
+	const char *buffer_usage_threshold_value = NULL;
+	const char *nr_expected_notifications_string = NULL;
+
+	session_name = argv[1];
+	channel_name = argv[2];
+	domain_type_string = argv[3];
+	buffer_usage_type_string = argv[4];
+	buffer_usage_threshold_type = argv[5];
+	buffer_usage_threshold_value = argv[6];
+	nr_expected_notifications_string = argv[7];
+
+	/* Parse arguments */
+	/* Domain type */
+	if (!strcasecmp("LTTNG_DOMAIN_UST", domain_type_string)) {
+		domain_type = LTTNG_DOMAIN_UST;
+	}
+	if (!strcasecmp("LTTNG_DOMAIN_KERNEL", domain_type_string)) {
+		domain_type = LTTNG_DOMAIN_KERNEL;
+	}
+	if (domain_type == LTTNG_DOMAIN_NONE) {
+		printf("error: Unknown domain type");
+		goto error;
+	}
+
+	/* Buffer usage condition type */
+	if (!strcasecmp("low", buffer_usage_type_string)) {
+		buffer_usage_type = LTTNG_CONDITION_TYPE_BUFFER_USAGE_LOW;
+	}
+	if (!strcasecmp("high", buffer_usage_type_string)) {
+		buffer_usage_type = LTTNG_CONDITION_TYPE_BUFFER_USAGE_HIGH;
+	}
+	if (buffer_usage_type == LTTNG_CONDITION_TYPE_UNKNOWN) {
+		printf("error: Unknown condition type");
+		goto error;
+	}
+
+	/* Ratio or bytes ? */
+	if (!strcasecmp("bytes", buffer_usage_threshold_type)) {
+		is_threshold_ratio = false;
+		sscanf(buffer_usage_threshold_value, "%" SCNu64, &threshold_bytes);
+	}
+
+	if (!strcasecmp("ratio", buffer_usage_threshold_type)) {
+		is_threshold_ratio = true;
+		sscanf(buffer_usage_threshold_value, "%lf", &threshold_ratio);
+	}
+
+	/* Number of notification to expect */
+	sscanf (nr_expected_notifications_string, "%d", &nr_expected_notifications);
+
+	return 0;
+error:
+	return 1;
+}
+
+int main(int argc, char **argv)
+{
+	int ret = 0;
+
+	enum lttng_condition_status condition_status;
+	enum lttng_notification_channel_status nc_status;
+	struct lttng_notification_channel *notification_channel = NULL;
+	struct lttng_condition *condition = NULL;
+	struct lttng_action *action = NULL;
+	struct lttng_trigger *trigger = NULL;
+
+	/* No buffering on stdout */
+	setbuf(stdout, NULL);
+
+	if (argc < 8) {
+		assert("Missing arguments for tests");
```

Comment:
This will always be true. printf() and exit with error. Same elsewhere (other similar `assert()`).